cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -44,6 +44,7 @@ var cloudproviders = []string{
 	"aws",
 	"azure",
 	"cloudstack",
+	"external",
```

Comment:
> I thought --apiserver-cert-extra-sans would address the problem at first but I'm not sure anymore

yeah, this is a kubelet serving cert issue, not an apiserver issue, and it's not a kubeadm-specific problem. we have two alpha-level things (external cloud providers and tls server bootstrapping) that do not work together. for them to work, adjustments have to be made to the kubelet to allow reversing the data flow of the kubelet addresses (instead of kubelet providing them to the node status, it has to consume them from the node status)